kod/object/active/holder/nomoveon/battler/monster/towns.kod

```diff
@@ -32,6 +32,13 @@ properties:
    piFace_translation = PT_GRAY_TO_SKIN1
    piHair_translation = PT_HAIR_DKBROWN
 
+   % Tracking the last time this NPC welcomed a player
+   piLastWelcomeTime = 0
+
+   % Minimum interval (in seconds) between NPC welcome messages to prevent excessive 
+   % chatter when a player enters their room.
+   piWelcomeTimeInterval = 120
```

Comment:
Not sure if this should be a class variable or a property, as there's typically only one of each NPC type.  Your thoughts?